### PR TITLE
remove ./node_modules/.bin from npm scripts

### PR DESCRIPTION
**Pull request by collinsauve**
Fixes issue #74, which broke linting and tests on Windows.

### CODE REVIEW

**Comment by paulius005** [comment]
🎉

File: package.json [hunk 1]
```diff
@@ -4,10 +4,10 @@
   "description": "A set of utilities for building Redux applications in Google Chrome Extensions.",
   "main": "lib/index.js",
   "scripts": {
-    "lint": "./node_modules/.bin/eslint src/**/*.js && ./node_modules/.bin/eslint test/**/*.js",
-    "prepublish": "./node_modules/.bin/babel src --out-dir lib",
-    "pretest": "./node_modules/.bin/babel src --out-dir lib",
-    "test": "npm run-script lint && ./node_modules/.bin/mocha --compilers js:babel-core/register"
+    "lint": "eslint src/**/*.js && eslint test/**/*.js",
```
Comment (paulius005) [inline package.json]: does this work if eslint is not installed globally?

**Comment by vhmth** [comment]
I like the dot-delimiter because it makes this repo self-contained. Right now this config makes the assumption that the user has exported `node_modules` from every subdir on their path. Does this not work with `./` as separate commands on Windows?

File: package.json (hunk 1 above)
Comment (paulius005) [inline package.json]: to get this to work I normally have to modify my path to include `:node_modules/.bin/`

Comment (paulius005) [inline package.json]: I'm more in favor of the style of #75

**Comment by paulius005** [comment]
@vhmth looks like it does. Take a look @ #75

**Comment by vhmth** [comment]
My vote would be for `./` as separate commands for the Windows case.

**Comment by collinsauve** [comment]
Here's a [comment about this problem](https://github.com/npm/npm/issues/4040#issuecomment-27159695) by isaacs (inventor of npm):

> Also, never ever write ./node_modules/.bin/ in your scripts. Seriously, that should just throw. What if the dep is somewhere else, installed globally, or as a parent's dep? npm puts ./node_modules/.bin in the PATH environ for a reason. Do not tightly couple to non-portable platform-specific paths! </rant>

Note that he says "npm puts ./node_modules/.bin in the PATH".  I did not need to install anything globally to get this to work.

**Comment by collinsauve** [comment]
> Does this not work with ./ as separate commands on Windows?

It works, but (as well as taking isaacs comment) I found it kind of hacky in that it creates scripts that really would only be used as building blocks for other scripts, and never run independently.

File: package.json (hunk 1 above)
Comment (collinsauve) [inline package.json]: I don't have `eslint` in my path, and I don't think I have it installed in npm global (`npm ls -g | grep eslint` is empty).   

`npm lint` runs for me on windows (as do the other scripts).

**Comment by vhmth** [comment]
https://www.useloom.com/share/e39a52d3b722475294409f092ef21e44

**Comment by collinsauve** [comment]
:) Thanks for the insight.   I don't know this kind of history / insight into npm - I wasn't aware it was this complex and debated.

I actually generally work in C# (just using JS for a chrome extension), so when you talk about "too much dynamic linking" going on in node - I definitely hear that! 

So you've convinced me - since you value predictability and isolation over my arbitrary "cleanliness" of the scripts, #75 would be the way to go.

**Comment by vhmth** [comment]
@collinsauve cool glad I could convince you! There is no right or wrong answer here though. Totally game to leave it up to @tshaddix since he has to sign off on stuff like this too. :-)

**Comment by tshaddix** [comment]
@vhmth I'm happy to go with whatever you see as the best option here :)

**Comment by tshaddix** [comment]
Closing this in favor of #83

**Pull request closed without merging**